### PR TITLE
[feature-wip](unique-key-merge-on-write) update counter, DSIP-018

### PR DESCRIPTION
**Pull request by zhannngchen**
# Proposed changes

Issue Number: close #xxx

## Problem Summary:
The follow up patch for #11136 

## Checklist(Required)

1. Type of your changes:
    - [ ] Improvement
    - [ ] Fix
    - [x] Feature-WIP
    - [ ] Feature
    - [ ] Doc
    - [ ] Refator
    - [ ] Others: 
2. Does it affect the original behavior: 
    - [ ] Yes
    - [x] No
    - [ ] I don't know
3. Has unit tests been added:
    - [ ] Yes
    - [ ] No
    - [x] No Need
4. Has document been added or modified:
    - [ ] Yes
    - [ ] No
    - [x] No Need
5. Does it need to update dependencies:
    - [ ] Yes
    - [x] No
6. Are there any changes that cannot be rolled back:
    - [ ] Yes
    - [x] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/olap/rowset/segment_v2/segment_iterator.cpp [hunk 1]
```diff
@@ -156,7 +156,9 @@ Status SegmentIterator::_init(bool is_vec) {
     // Remove rows that have been marked deleted
     if (_opts.delete_bitmap.count(segment_id()) > 0 &&
         _opts.delete_bitmap[segment_id()] != nullptr) {
+        size_t pre_size = _row_bitmap.cardinality();
         _row_bitmap -= *(_opts.delete_bitmap[segment_id()]);
```
Comment (BiteTheDDDDt) [inline be/src/olap/rowset/segment_v2/segment_iterator.cpp:160]: Maybe we can add a CHECK `pre_size > _row_bitmap.cardinality()`

Comment (zhannngchen) [reply be/src/olap/rowset/segment_v2/segment_iterator.cpp:160]: Why do we need such a CHECK here? Do you want to ensure that the delete_bitmap is not empty?

Comment (BiteTheDDDDt) [reply be/src/olap/rowset/segment_v2/segment_iterator.cpp:160]: Sorry,  I read it wrong, I read `-=` as `=` 😥

**Review by BiteTheDDDDt — DISMISSED** [review]
LGTM

**Comment by BiteTheDDDDt** [comment]
BTW, is this pr related to #11191?

File: be/src/olap/rowset/segment_v2/segment_iterator.cpp (hunk 1 above)
Comment (zhannngchen) [reply be/src/olap/rowset/segment_v2/segment_iterator.cpp:160]: LOL, thanks for your review~

**Comment by zhannngchen** [comment]
> BTW, is this pr related to #11191?

no，see the PR desc, it's a follow up patch for #11136, part of WIP feature of DSIP-018

**Review by kpfly — APPROVED** [review]
LGTM

**Review by dataroaring — APPROVED** [review]
LGTM

**Pull request merged**